### PR TITLE
Registration: Add authentication.

### LINKED ISSUE #4: Enable migrations and metadata persisting
In order to enable automatic DB migrations and metadata applying on initialization we need to:
- [x] Creating scripts for generating migrations and applying migrations
- [x] Applying said migrations automatically on load

More info in https://docs.hasura.io/1.0/graphql/manual/migrations/index.html

### PR DESCRIPTION
**Pull request by wildeyes**
Closes #4.

### CODE REVIEW

File: services/handlers/authentication.ts [hunk 1]
```diff
@@ -1,86 +1,155 @@
-import logger from '../utilities/logger'
```
Comment (shakedlokits) [inline services/handlers/authentication.ts]: Remove formatting changes. Does not relate to PR.

File: services/handlers/dummy.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { resolverWithAuth } from './authentication'
```
Comment (shakedlokits) [inline services/handlers/dummy.ts]: Remove dummy endpoint, I this is not required other then personal testing.

File: services/index.ts [hunk 3]
```diff
@@ -1,38 +1,79 @@
-import * as express from 'express'
```
Comment (shakedlokits) [inline services/index.ts]: Remove formatting changes here as well.

File: services/index.ts [hunk 4]
```diff
@@ -1,38 +1,79 @@
...
-
-    if (isInvalidated && isInternal && !isIntrospection) {
-        res.status(403).json({
-            errors: ['Unauthorized access']
-        })
-    }
-    next()
+  if (!req.body['query']) {
+    return res.status(400).json({
+      errors: ['Document must not be empty'],
```
Comment (shakedlokits) [inline services/index.ts]: I don't think we need to tell the client that a `/graphql` endpoint requires a payload.

Comment (shakedlokits) [reply services/index.ts]: And if so it should have been `418`

File: services/schema.graphql [hunk 5]
```diff
@@ -1,26 +1,31 @@
 input MailInput {
```
Comment (shakedlokits) [inline services/schema.graphql]: Remove formatting changes.

File: docker-compose.yml [hunk 6]
```diff
@@ -13,25 +13,29 @@ services:
       - hasura
     build:
       context: ./registration
-      dockerfile: Dockerfile
+      dockerfile: Dockerfile.dev
```
Comment (shakedlokits) [inline docker-compose.yml]: You shouldn't change the production environment values and it also does not relate the the PR.

**Review by shakedlokits — CHANGES_REQUESTED** [review]

File: services/handlers/authentication.ts (hunk 1 above)
Comment (wildeyes) [reply services/handlers/authentication.ts]: removed formatting.

File: services/handlers/dummy.ts (hunk 2 above)
Comment (wildeyes) [reply services/handlers/dummy.ts]: removed.

File: services/index.ts (hunk 3 above)
Comment (wildeyes) [reply services/index.ts]: removed formatting.

File: services/index.ts (hunk 4 above)
Comment (wildeyes) [reply services/index.ts]: removed.

File: services/schema.graphql (hunk 5 above)
Comment (wildeyes) [reply services/schema.graphql]: removed formatting.

**Pull request closed without merging**